File: src/main/java/kr/pe/ecmaxp/openpie/Result.java
package kr.pe.ecmaxp.openpie;

public class Result
{
    public final Call call;
    public final Object[] args;
    public final Throwable error;

    public Result(Call call, Object[] args)
    {
        this.call = call;
        this.args = args;
        this.error = null;
    }

    public Result(Call call, Throwable throwable)
    {
        this.call = call;
        this.args = null;
        this.error = throwable;
    }
}

File: src/main/java/kr/pe/ecmaxp/openpie/arch/types/Result.kt
package kr.pe.ecmaxp.openpie.arch.types

import java.util.*

class Result(val invoke: Invoke, val args: Array<Any?>? = null, val error: Throwable? = null) {
    override fun toString(): String {
        return "Result(invoke=$invoke, args=${Arrays.toString(args)}, error=$error)"
    }
}
